posthog/api/survey.py

```diff
@@ -937,7 +937,15 @@ def surveys(request: Request):
             ),
         )
 
-    return cors_response(request, JsonResponse(get_surveys_response(team)))
+    if team.surveys_opt_in:
+        return cors_response(request, JsonResponse(get_surveys_response(team)))
+    else:
+        response = {
+            "surveys": [],
+            "survey_config": None,
+        }
+
+        return cors_response(request, JsonResponse(response))
```

Comment:
this change I am not so sure since if somebody calls the survey APIs, I expect them to know what they are doing.
most likely i'd need to do the same chance in the other possible APIs.
one of them is used by our frontend (all CRUD operations), and another one is used by the SDK.

Reply:
FE is gonna use this API so we should not do this here